### PR TITLE
Create ListItem component for Dropdown component

### PR DESCRIPTION
**Pull request by Rheniery**
If applied, this pull request will create the ListItem component with the simple and checkbox types for the Dropdown component.

### Other minor changes:
- Change Dropdown component to accept the ListItem component

### Card Link:
N/A

### Design Expected Screenshot
Seleção_144
Seleção_143

### Implementation Screenshot or GIF
Seleção_147
Seleção_150

### Notes:
N/A

### CODE REVIEW

File: src/components/Dropdown/index.tsx [hunk 1]
```diff
@@ -1,66 +1,97 @@
...
 import { useSelect } from 'downshift';
 import ClassNames from 'classnames';
+import isEqual from 'lodash.isequal';
 
 import styles from '@/components/Dropdown/Dropdown.css';
 
 import IconGear from '@/assets/images/icons/web/gear.svg';
 import IconArrowUp from '@/assets/images/icons/web/arrow-up.svg';
 import IconArrowDown from '@/assets/images/icons/web/arrow-down.svg';
+import ListItem from '../ListItem';
```
Comment (diogoribeiro) [inline src/components/Dropdown/index.tsx:11]: let's avoid using relative imports

File: src/components/Dropdown/Dropdown.css [hunk 2]
```diff
@@ -175,9 +172,8 @@
   filter: var(--dropdown-icon-filter);
 }
 
-.dropdown-icon-gear {
+.dropdown-icon-cog {
```
Comment (kaiomagalhaes) [inline src/components/Dropdown/Dropdown.css:175]: Please rename this to `dropdown-icon-gear`

File: src/components/ListItem/CheckBoxListItem/index.tsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import DefaultListItem, { Props } from '@/components/ListItem/DefaultListItem'; // eslint-disable-line no-unused-vars
```
Comment (diogoribeiro) [inline src/components/ListItem/CheckBoxListItem/index.tsx:2]: we're using both items here, do we really need `eslint-disable-line no-unused-vars`?

File: src/components/Dropdown/Dropdown.stories.tsx [hunk 4]
```diff
@@ -85,5 +87,87 @@ LabelDropdown.args = {
       label: 'Option 2',
       value: '2'
     }
-  ]
+  ],
+  getItemLabel: item => item.label
+};
+
+export const CustomIconDropdown = Template.bind({});
```
Comment (kaiomagalhaes) [inline src/components/Dropdown/Dropdown.stories.tsx:94]: Please review the names here

File: src/components/Dropdown/index.tsx [hunk 5]
```diff
@@ -1,138 +1,176 @@
...
-    highlightedIndex
+    highlightedIndex,
+    isOpen
   } = useSelect<T>({
     items: options,
     selectedItem: selected,
     onSelectedItemChange
   });
 
+  const largeSize = 'large';
```
Comment (kaiomagalhaes) [inline src/components/Dropdown/index.tsx:67]: Please move this to line 12 and reuse it on line 15

Comment (kaiomagalhaes) [reply src/components/Dropdown/index.tsx:67]: Same for the icon

File: src/components/Dropdown/index.tsx [hunk 6]
```diff
@@ -1,138 +1,176 @@
...
+  const IconCategory = 'icon';
```
Comment (kaiomagalhaes) [inline src/components/Dropdown/index.tsx:68]: please use Capital case only for components

**Review by arathjz — APPROVED** [review]

File: src/components/Dropdown/index.tsx [hunk 7]
```diff
@@ -1,138 +1,176 @@
...
       </div>
     </div>
   );
-}
+};
 
 Dropdown.defaultProps = {
   category: 'simple',
-  id: 'ads-dropdown',
   size: 'large',
```
Comment (kaiomagalhaes) [inline src/components/Dropdown/index.tsx:169]: please use the new large variable here

File: src/components/ListItem/DefaultListItem/index.tsx [hunk 8]
```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import classNames from 'classnames';
+
+import styles from '@/components/ListItem/DefaultListItem/DefaultListItem.css';
+
+export interface Props<T> {
```
Comment (kaiomagalhaes) [inline src/components/ListItem/DefaultListItem/index.tsx:6]: sort these alphabetically

File: src/components/ListItem/DefaultListItem/index.tsx [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+      className={classNames(className, styles['list-item'], variablesClassName)}
+      role='button'
+      aria-pressed='false'
+      tabIndex={0}
+    >
+      <div>{getIcon && (size === 'large' ? getIcon(item, 20) : getIcon(item, 13))}</div>
+      <span className={classNames(styles['list-item-span'], styles[sizeSpanClass])}>
+        {getLabel(item)}
+      </span>
+      <span>{getSelectedMarker && getSelectedMarker(item)}</span>
```
Comment (kaiomagalhaes) [inline src/components/ListItem/DefaultListItem/index.tsx:43]: this will always render even if no function is passed because of your default value

**Review by kaiomagalhaes — CHANGES_REQUESTED** [review]

File: src/components/ListItem/CheckBoxListItem/index.tsx (hunk 3 above)
Comment (Rheniery) [reply src/components/ListItem/CheckBoxListItem/index.tsx:2]: There is a problem in the TS that it doesn't recognize the Props variable.
Captura de tela de 2020-12-01 17-09-58

**Review by kaiomagalhaes — APPROVED** [review]

**Pull request merged**